packages/neo-one-node/src/FullNode.ts

```diff
@@ -34,7 +38,23 @@ export class FullNode {
             reject(error);
           }
         };
-        this.mutableSubscription = fullNode$(this.options).subscribe({
+        this.mutableSubscription = combineLatest(
+          fullNode$(this.options),
+          this.options.environment.haltOnSync
+            ? interval(5000).pipe(
+                map(async () => {
+                  try {
+                    const response = await fetch(`http://localhost:${port}/ready_health_check`);
+                    if (response.status === 200) {
+                      await this.stop(true);
```

Comment:
Move this into it's own `switchMap` otherwise there's a race condition. (Try to figure out why)